java/src/jmri/managers/configurexml/DefaultSignalMastManagerXml.java

```diff
@@ -156,6 +156,23 @@ public boolean load(Element shared, Element perNode) {
             }
         }
 
+        list = shared.getChildren("matrixsignalmast");
+        if (list != null) {
+            for (int i = 0; i < list.size(); i++) {
+                Element e = list.get(i);
+                String adapterName = e.getAttribute("class").getValue();
+                log.debug("load via " + adapterName);
+                try {
+                    XmlAdapter adapter = (XmlAdapter) Class.forName(adapterName).newInstance();
+                    // and do it
+                    adapter.load(e, null);
+                } catch (Exception ex) {
+                    log.error("Exception while loading " + e.getName() + ":" + ex);
+                    ex.printStackTrace();
```

Comment:
Kind of. I meant replace:
```
log.error("Exception while loading " + e.getName() + ":" + ex);
ex.printStackTrace();
```
with
```
log.error("Exception while loading {}: {}", e.getName(), ex, ex);
```
because ```log.error("Exception: {}", ex)``` shows just "Exception: the exception's message", while ```log.error("Exception: {}", ex, ex)``` (no placeholder {} for the 2nd ex) shows "Exception: the exceptions's message" followed by the stack trace. The org.slf.Logger trace/debug/info/warn/error methods will accept an Exception as the last argument without a matching placeholder to dump the stack trace to the log.